lib/cryptoOperations.js

```diff
@@ -0,0 +1,28 @@
+const { subtle } = globalThis.crypto;
+import keytar from 'keytar';
+
+const SERVICE_NAME = 'filecoin-station-core';
+
+export async function getStationId() {
+  const storedKeys = await keytar.getPassword(SERVICE_NAME, 'station_id');
```

Comment:
I would like to discuss where to store the private key.

About the [`keytar`](https://github.com/atom/node-keytar) library:
- It is also used in Station Desktop, for storing the wallet seed phrase
- It's deprecated
- On some Windows versions, it's not possible for the user to manually read out the value, without the app offering a function for it

Station Core has two main audiences:
- Inside Station Desktop, running on all 3 major operating systems
- Inside Docker, on some linux flavor

I believe that we want to disallow exporting the private key, as it shouldn't be reused across machines, and doesn't need to be backed up. Therefore, the Windows limitation with keytar is no problem. We can't prevent exporting the private key, since it needs to be stored _somewhere_. But, it's good for us if it isn't easy either.

I believe therefore that keytar _is_ the right solution, as it stores it in a way that's encrypted with the current user's credentials, and isn't as accessible as say a file on disk. I would like to eventually find a replacement for `keytar` that is maintained, but for now we don't have a problem.

One concern I have is that keytar on macOS will ask for the user's password, before allowing operation. Since we use it in Station Desktop and now soon Station Core, we need to prevent asking for the password twice. I think that won't happen, but we need to verify it. I tried running the current PR, but got `[SyntaxError]: Unsupported key usage for an Ed25519 key`. Since it got this far, it seems to indicate that no password confirmation is required. Let's verify this inside Station Desktop once the PR is in a working shape.

Reply:
This all makes sense to me!

> I believe that we want to disallow exporting the private key

Do you mean we should do something extra to disallow exporting the private key here, or just that keytar is a good solution for now because it makes this nontrivial?

> Let's verify this inside Station Desktop once the PR is in a working shape.

Do you recommend that I try to get my own dev application working on my machine to do this spot-checking, or that you and Miro do this, and I focus on unit tests and such?

Reply:
> Do you mean we should do something extra to disallow exporting the private key here, or just that keytar is a good solution for now because it makes this nontrivial?

The latter 👍 

> Do you recommend that I try to get my own dev application working on my machine to do this spot-checking, or that you and Miro do this, and I focus on unit tests and such?

I'm happy to do it, just ping me 🙏